3634/3/app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,22 @@
+require 'pry'
+
+class RestaurantsController < ApplicationController
+  before '/restaurants/:id' do
+    redirect '/notfound' if restaurant_not_exist?
```

Comment:
don't use a filter for this. Sinatra has great tools to handle exceptions
http://sinatrarb.com/intro.html
Error Handling - section
there is a cool example for you.
Add global handling for `ActiveRecord::RecordNotFound` error. And render :not_found page for this.
ActiveRecord method `find` raise the error when couldn't find the record with passed id.